tuning/trainercontroller/controllermetrics/history_based_metrics.py

```diff
@@ -0,0 +1,97 @@
+# Copyright The IBM Tuning Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# SPDX-License-Identifier: Apache-2.0
+# https://spdx.dev/learn/handling-license-info/
+
+# Standard
+from collections import deque
+from typing import Any
+
+# Third Party
+from transformers import TrainerState
+from transformers.utils import logging
+
+# Local
+from tuning.trainercontroller.controllermetrics.metricshandler import MetricHandler
+
+logger = logging.get_logger(__name__)
+
+
+class HistoryBasedMetric(MetricHandler):
+    """Implements the controller metric which evaluates loss-per-step"""
+
+    _METRICS_KEY = "metrics"
+    _TRAINING_LOSS_KEY = "loss"
+    _WINDOW_SIZE = "window-size"
+
+    def __init__(self, **kwargs):
+        """Initializes the metric handler, by registering the event \
+            list and arguments with base handler.
+
+        Args:
+            kwargs: List of arguments (key, value)-pairs
+        """
+        self._window_size = kwargs.get("window-size")
+        self._window = {
+            self._TRAINING_LOSS_KEY: deque(),
+            self._METRICS_KEY: deque(),
+            self._WINDOW_SIZE: self._window_size,
+        }
+        if self._window_size is None:
```

Comment:
Use function default params to initialize

Reply:
Addressed.